### ISSUE TITLE
Support for multi-line repl input

### ISSUE THREAD

**Issue by prologic** [NicoNex/tau#17]
Something I noticed, the repl doesn't support multi-line inputs:

For example:

```#!tau
>>> div = fn(n, d) {
closure[0xc000076140]
>>> if d == 0 {
undefined variable d
>>> return error("zero division error")
runtime error: runtime error: index out of range [-1]
>>> }
no parse prefix function for '}' found
>>> n / d
undefined variable n
>>> }
```

it should 😅 IMO

**Comment by prologic** [comment]
<img width="1242" alt="Screen Shot 2022-07-27 at 21 00 24" src="https://user-images.githubusercontent.com/1290234/181231563-1abb58f7-624d-4365-bbed-c4d8c816d7b2.png">

**Comment by NicoNex** [comment]
That's definitely in my TODO list. :D

**Comment by prologic** [comment]
Thinking about this some more and other things I'm sort-of involved in... There's a couple of libraries that _could_ help here and also refactor a lot of the custom REPL handling out of Tau:

- https://github.com/abiosoft/ishell
- https://github.com/peterh/liner

**Comment by NicoNex** [comment]
I checked those out and I don't think they can help with that, and they look less simple than the one I'm currently using.
For the multiline I think I can simply check if the last character before the `\n` is `{` or maybe some others that I can't think of rn and if so keep on accepting lines until `\n\n` is pressed.

We could even add a counter that gets incremented everytime `{` is at the end of the line and decremented when `}` is at the end of the line even though I think that just waiting for `\n\n` might be enough.

**Comment by NicoNex** [comment]
```
>>> test = fn(a, b) {
...     println("test multiline", a, b)
... }
... 
closure[0xc000066180]
>>> test(1, 2)
test multiline 1 2
>>> 
```

Resolved with 66e9d3968ce1866c190c0c84ca3d4db35fa98117.

**Comment by prologic** [comment]
Very nice! 👌
